plugins/observer/main.go

```diff
@@ -0,0 +1,325 @@
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	bq, err := bigquery.NewClient(ctx, cfg.BigQuery.ProjectID, option.WithCredentialsFile(cfg.BigQuery.CredentialFile))
+	if err != nil {
+		logrus.WithError(err).Fatal("Error getting BigQuery client.")
+	}
+	err = setupBigQuery(ctx, bq, cfg.BigQuery.Dataset)
+	if err != nil {
+		logrus.WithError(err).Fatal("Error setting up BigQuery session.")
+	}
+
+	mux := http.NewServeMux()
+	externalplugins.ServeExternalPluginHelp(mux, log, helpProvider)
+	httpServer := &http.Server{Addr: ":" + strconv.Itoa(o.port), Handler: mux}
+
+	health := pjutil.NewHealth()
+	health.ServeReady()
+	defer interrupts.WaitForGracefulShutdown()
+	defer cancel()
+
+	tbl := bq.Dataset(cfg.BigQuery.Dataset).Table(tableName)
+
+	var wg sync.WaitGroup
+	defer wg.Wait()
+	for orgRepo, repoCfg := range cfg.OrgRepos {
+		for projectNumber, prj := range repoCfg.Projects {
+			wg.Add(1)
+			go observeRepo(ctx, &wg, tbl, githubClient, orgRepo, projectNumber, prj)
+		}
+	}
+
+	interrupts.ListenAndServe(httpServer, 5*time.Second)
+}
+
+type ProjectColumnEntry struct {
+	Project   string
+	Issue     int
+	Title     string
+	Column    string
+	Timestamp time.Time
+	Assignees []string
+}
+
+const tableName = "project_col_entry"
+
+func setupBigQuery(ctx context.Context, client *bigquery.Client, dataset string) error {
+	schema, err := bigquery.InferSchema(ProjectColumnEntry{})
+	if err != nil {
+		return err
+	}
+	ds := client.Dataset(dataset)
+	err = ds.Create(ctx, nil)
+	if err != nil && !isAlreadyExists(err) {
+		return err
+	}
+
+	tbl := ds.Table(tableName)
+	err = tbl.Create(ctx, &bigquery.TableMetadata{Schema: schema})
+	if err != nil && !isAlreadyExists(err) {
+		return err
+	}
+
+	return nil
+}
+
+func isAlreadyExists(err error) bool {
+	var gerr *googleapi.Error
+	if errors.As(err, &gerr) {
+		if gerr.Code == http.StatusConflict {
+			return true
+		}
+	}
+	return false
+}
+
+// {
+// 	organization(login: "gitpod-io") {
+// 	  project(number: 3) {
+// 		columns(first: 100) {
+// 		  nodes {
+// 			name
+// 			cards(first: 100) {
+// 			  nodes {
+// 				content {
+// 				  ... on Issue {
+// 					id
+// 					number
+// 					title
+// 				  }
+// 				  ... on PullRequest {
+// 					id
+// 					number
+// 					title
+// 				  }
+// 				}
+// 			  }
+// 			}
+// 		  }
+// 		}
+// 	  }
+// 	}
+//   }
+
+type queryIssue struct {
+	Number    githubql.Int
+	Title     githubql.String
+	Assignees struct {
+		Nodes []struct {
+			Login githubql.String
+		}
+	} `graphql:"assignees(first: 5)"`
+}
+
+type projectQuery struct {
+	Organisation struct {
+		Project struct {
+			Name    githubql.String
+			Columns struct {
+				Nodes []struct {
+					Name  string
+					Cards struct {
+						Nodes []struct {
+							Content struct {
+								PR    queryIssue `graphql:"... on PullRequest"`
+								Issue queryIssue `graphql:"... on Issue"`
+							}
+						}
+					} `graphql:"cards(first: 100)"`
```

Comment:
Any reason to limit to 100 cards? The „Done“ column could be quite full.

Reply:
the done column is in the ignore list :)

Reply:
basically once the cards drop out of the query they're considered done if they were in "Await Deployment" before